tools/x-docs-2/src/components/sidebar/module-list.jsx

```diff
@@ -1,17 +1,65 @@
 import React from 'react';
 import { Link } from 'gatsby';
 
-const List = ({ data, title }) => (
+import GithubSlugger from 'github-slugger';
+
+const createUrlFromName = (name) => {
+	const slugger = new GithubSlugger();
+	// As slugger checks if it is not duplicating created urls, like this:
+	// const slugger = new GithubSlugger();
+	// slugger.slug('url one') // url-one
+	// slugger.slug('url one') // url-one-2
+	// therefore there is a need to create a new class instance every time when the function is applied
+	return slugger.slug(name)
+}
+
+const reactToClick = (e, link) => {
+	e.preventDefault();
+	document.querySelector(`#${link}`).scrollIntoView({
```

Comment:
Niiice, but we can go further! The event itself gives us a reference to the link, and the link gives us the ID =]

https://developer.mozilla.org/en-US/docs/Web/API/Event/target
https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/hash

Reply:
magic!